lib/reducer.ts

```diff
@@ -102,13 +102,34 @@ export const introspectionFieldReducerGenerator: (
   return introspectionFieldReducer
 }
 
-// ENUM type defaults will not JSON.parse correctly, so if it is an ENUM then don't
-// try to do that.
+// ENUM type defaults will not JSON.parse correctly, so we have to do some work
 // TODO: fix typing here
 export const resolveDefaultValue = (curr: any) => {
-  return isIntrospectionEnumType(curr.type)
-    ? curr.defaultValue
-    : JSON.parse(curr.defaultValue)
+  let type = curr.type
+  let isList = false
+
+  // Dig out the underlying type in case it's a LIST or a NON_NULL or both
+  while (true) {
```

Comment:
it would be better to have a recursive `resolveDefaultValue()` instead of this `while(true) {}`

Reply:
Hmm, there are few reasons that I avoided that:
1. While the information about the underlying type may be nested deep down, once we find it the value we're going to be working with is on the outer `curr` that was originally passed in...specifically `curr.defaultValue`
2. While we're going through the loop/recursion, we need to take note as to whether the thing turns out to be a `LIST` or not

Here's what I'd probably write:

```js
function digType(type, isList) {
  if (isIntrospectionListTypeRef(type)) {
    return digType(type.ofType, true)
  } else if (isNonNullIntrospectionType(type)) {
      return digType(type.ofType, isList)
  }
  return [type, isList]
}

function resolveDefaultValue (curr) {
  const [type, isList] = digType(curr.type, false)
  ...
}
```

Sure, it could be done using recursion, but it feels like it'd a bit forced/contrived to me...would you agree?

I'll do whatever, though...

Reply:
@charlypoly I just tried to do this, but TS was complaining about types, and I don't know enough about TypeScript to get it to work.

Could you maybe merge as-is and then fix to your liking? I just don't know enough about TypeScript and I'm sure for you it'd be a quick change

Reply:
> @charlypoly I just tried to do this, but TS was complaining about types, and I don't know enough about TypeScript to get it to work.
> 
> Could you maybe merge as-is and then fix to your liking? I just don't know enough about TypeScript and I'm sure for you it'd be a quick change

bump @charlypoly ... I don't have the skills to do your suggested update...could you handle it after merging this PR?